tree.go

```diff
@@ -1266,3 +1266,155 @@ func ToDot(root VerkleNode) string {
 	root.Commit()
 	return fmt.Sprintf("digraph D {\n%s}", root.toDot("", ""))
 }
+
+type SerializedNode struct {
+	Node            VerkleNode
+	CommitmentBytes [32]byte
+	SerializedBytes []byte
+}
+
+// BatchSerialize is an optimized serialization API when multiple VerkleNodes serializations are required, and all are
+// available in memory.
+func (n *InternalNode) BatchSerialize() ([]SerializedNode, error) {
+	// Commit to the node to update all the nodes commitments.
+	n.Commit()
+
+	// Collect all nodes that we need to serialize.
+	nodes := make([]VerkleNode, 0, 1024)
+	nodes = n.collectNonHashedNodes(nodes)
```

Comment:
We collect in `nodes` all touched nodes from the tree (i.e: non-hash nodes; that is: present internal and leaf nodes).